src/profile/components/basicData/BasicData.tsx

```diff
@@ -1,28 +1,172 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useTranslation } from 'react-i18next';
+import { Field, Formik, FormikProps, Form } from 'formik';
+import { TextInput } from 'hds-react';
+import to from 'await-to-js';
 
 import commonFormStyles from '../../../common/cssHelpers/form.module.css';
 import LabeledValue from '../../../common/labeledValue/LabeledValue';
 import ProfileSection from '../../../common/profileSection/ProfileSection';
 import { getFormFields } from '../../helpers/formProperties';
-import { basicDataType, BasicDataValue } from '../../helpers/editData';
-import { useProfileDataEditor } from '../../hooks/useProfileDataEditor';
+import {
+  basicDataType,
+  BasicDataValue,
+  EditDataValue,
+} from '../../helpers/editData';
+import {
+  ActionListener,
+  saveTypeToAction,
+  useProfileDataEditor,
+} from '../../hooks/useProfileDataEditor';
+import { basicDataSchema } from '../../../common/schemas/schemas';
+import { getFieldError, getIsInvalid } from '../../helpers/formik';
+import useNotificationContent from '../editingNotifications/useNotificationContent';
+import EditingNotifications from '../editingNotifications/EditingNotifications';
+import EditButtons, { ActionHandler } from '../editButtons/EditButtons';
+import FormButtons from '../formButtons/FormButtons';
+import SaveIndicator from '../saveIndicator/SaveIndicator';
+
+type FormikValues = BasicDataValue;
 
 function BasicData(): React.ReactElement | null {
-  const { editDataList } = useProfileDataEditor({
+  const { editDataList, save, reset } = useProfileDataEditor({
     dataType: basicDataType,
   });
+  const [isEditing, setEditing] = useState(false);
   const { t } = useTranslation();
+  const {
+    content,
+    setErrorMessage,
+    setSuccessMessage,
+    clearMessage,
+  } = useNotificationContent();
   const testId = basicDataType;
 
   if (!editDataList || !editDataList[0]) {
```

Comment:
This seems to be the first row where return values from _useProfileDataEditor_ function call are being applied. I wonder if it could it increase readability if we make the function call _useProfileDataEditor_ straight above this line 45?

Reply:
No, because the hook "useNotificationContent " cannot be run conditionally.

Reply:
I meant that useProfileDataEditor function call would be above this condition, at line 44

Reply:
Ah... Done.